packages/common/src/codegen/render-solidity/common.ts

```diff
@@ -22,8 +23,11 @@ export function renderList<T>(list: T[], renderItem: (item: T, index: number) =>
  * Renders a comma-separated list of arguments for solidity functions, ignoring empty and undefined ones
  */
 export function renderArguments(args: (string | undefined)[]): string {
-  const filteredArgs = args.filter((arg) => arg !== undefined && arg !== "") as string[];
-  return internalRenderList(",", filteredArgs, (arg) => arg);
+  return args
+    .filter(isDefined)
+    .filter((arg) => arg !== "")
+    .map((arg) => arg.replace(/ calldata /, " memory "))
```

Comment:
ah I actually meant doing it for the libraries only. Will this affect other codegen?

Reply:
doesn't seem to, I think because we already use memory everywhere in tables

Reply:
ah, it did have an impact on other interfaces, so I updated to just do this in lib output